### PR TITLE
Remove obsolete code from Xaml sources

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

Implements #

### Additions made ###

* Adds 

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

File: src/Controls/src/Xaml/ApplyPropertiesVisitor.cs [hunk 1]
```diff
@@ -774,10 +774,8 @@ static bool TryAddToResourceDictionary(ResourceDictionary resourceDictionary, ob
 
 		void SetTemplate(ElementTemplate dt, INode node)
 		{
-#pragma warning disable 0612
-			((IDataTemplate)dt).LoadTemplate = () =>
+			((ElementTemplate)dt).LoadTemplate = () =>
```
Comment (rmarinho) [inline src/Controls/src/Xaml/ApplyPropertiesVisitor.cs:777]: we don't need this cast

**Review by rmarinho — COMMENTED** [review]
Why do we remove the tests ?

File: src/Controls/src/Xaml/ApplyPropertiesVisitor.cs (hunk 1 above)
Comment (mattleibow) [reply src/Controls/src/Xaml/ApplyPropertiesVisitor.cs:778]:
```suggestion
			dt.LoadTemplate = () =>
```

**Review by mattleibow — APPROVED** [review]
LGTM

**Pull request merged**